test/distributed/org/apache/cassandra/distributed/test/auth/AuthAuditLoggingTest.java

```diff
@@ -0,0 +1,391 @@
+                AuditLogEntry entry = auditLogEntries.poll();
+                assertThat(entry).isNotNull();
+                assertThat(entry.getHost().toString(false)).isEqualTo("/127.0.0.1");
+                assertThat(entry.getSource().toString(false)).isEqualTo("/127.0.0.1");
+                assertThat(entry.getUser()).isNull();
+                assertThat(entry.getType()).isEqualTo(LOGIN_ERROR);
+                assertThat(entry.getLogString()).matches(expectedLogStringRegex);
+            });
+        }
+    }
+
+    @Test
+    public void testMutualTlsAuthenticationSuccessfulAuth() throws Exception
+    {
+        Path clientKeystorePath = generateClientCertificate(null, tempFolder.getRoot(), CA);
+        CharSequence expectedLogStringRegex = "^user:cassandra_ssl_test\\|host:.*/127.0.0.1:\\d+\\|source:/127.0.0.1" +
+                                              "\\|port:\\d+\\|timestamp:\\d+\\|type:LOGIN_SUCCESS\\|category:AUTH" +
+                                              "\\|operation:LOGIN SUCCESSFUL\\|identity:spiffe://test.cassandra.apache.org/dTest/mtls$";
+
+        try (com.datastax.driver.core.Cluster c = JavaDriverUtils.create(CLUSTER, null, b -> b.withSSL(getSSLOptions(clientKeystorePath, truststorePath)));
+             Session session = c.connect())
+        {
+            session.execute("DESCRIBE KEYSPACES");
+
+            CLUSTER.get(1).runOnInstance(() -> {
+                // We should have events recorded for the control connection and the session connection
+                AuditLogEntry entry1 = ((InMemoryAuditLogger) AuditLogManager.instance.getLogger()).internalQueue().poll();
+                assertThat(entry1).isNotNull();
+                assertThat(entry1.getHost().toString(false)).matches(".*/127.0.0.1");
+                assertThat(entry1.getSource().toString(false)).isEqualTo("/127.0.0.1");
+                assertThat(entry1.getUser()).isEqualTo("cassandra_ssl_test");
+                assertThat(entry1.getType()).isEqualTo(LOGIN_SUCCESS);
+                assertThat(entry1.getLogString()).matches(expectedLogStringRegex);
+                AuditLogEntry entry2 = ((InMemoryAuditLogger) AuditLogManager.instance.getLogger()).internalQueue().poll();
+                assertThat(entry2).isNotNull();
+                assertThat(entry2.getHost().toString(false)).matches(".*/127.0.0.1");
+                assertThat(entry2.getSource().toString(false)).isEqualTo("/127.0.0.1");
+                assertThat(entry2.getUser()).isEqualTo("cassandra_ssl_test");
+                assertThat(entry2.getType()).isEqualTo(LOGIN_SUCCESS);
+                assertThat(entry2.getLogString()).matches(expectedLogStringRegex);
+            });
+        }
+    }
+
+    @Test
+    public void testMutualTlsAuthenticationFailedWithUntrustedCertificate() throws Exception
+    {
+        configureMutualTlsAuthenticator();
+        CharSequence expectedLogStringRegex = "^user:null\\|host:.*/127.0.0.1:\\d+\\|timestamp:\\d+" +
```

Comment:
This fails for me locally because source is present for me (also on `testMutualTlsAuthenticationFailedWithExpiredCertificate`):

```
java.lang.AssertionError: 
Expecting actual:
  "user:null|host:localhost/127.0.0.1:42421|source:/127.0.0.1|port:49366|timestamp:1717027629823|type:LOGIN_ERROR|category:AUTH|operation:LOGIN FAILURE; Empty client certificate chain"
to match pattern:
  "^user:null\|host:.*/127.0.0.1:\d+\|timestamp:\d+\|type:LOGIN_ERROR\|category:AUTH\|operation:LOGIN FAILURE; Empty client certificate chain.*$"
```

Reply:
ah interesting . let me dig into this a bit more

Reply:
addressed